dataplane/updater.h

```diff
@@ -669,4 +670,118 @@ class updater_array
 	object_type* pointer;
 };
 
+template<typename Address, typename InnerLpmType>
+class updater_vrf_lpm
+{
+public:
+	using stats_t = typename InnerLpmType::stats_t;
+	using UpdaterType = updater_lpm<Address, InnerLpmType>;
+
+	updater_vrf_lpm(const char* name,
+	                dataplane::memory_manager* memory_manager,
+	                tSocketId socket_id) :
+	        name_(name),
+	        memory_manager_(memory_manager),
+	        socket_id_(socket_id)
+	{
+	}
+
+	eResult init()
+	{
+		return eResult::success;
+	}
+
+	eResult insert(tVrfId vrf,
+	               const Address& ip_address,
+	               const uint8_t& mask,
+	               const uint32_t& value_id)
+	{
+		if (vrf >= YANET_RIB_VRF_MAX_NUMBER)
+		{
+			return eResult::invalidId;
+		}
+		if (updaters_[vrf] == nullptr)
+		{
+			std::string name = std::string(name_) + ".vrf" + std::to_string(vrf);
+			updaters_[vrf] = std::make_unique<UpdaterType>(name.c_str(), memory_manager_, socket_id_);
+			if (updaters_[vrf] == nullptr)
+			{
+				return eResult::errorAllocatingMemory;
+			}
```

Comment:
This won't work. `std::make_unique` **never** returns `nullptr` - either it succeeds, or it throws `std::bad_alloc`.

If we want to return an appropriate `eResult`, we need to use one of the `new` operator overloads that takes `std::nothrow` object:
```cpp
auto new_updater = std::unique_ptr<UpdaterType>(
    new(std::nothrow) UpdaterType(uniqueName.c_str(), memory_manager_, socket_id_)
);
```

Or you can define helper function in the `util` namespace:
```cpp
template<typename T, typename... Args>
std::unique_ptr<T> make_unique_nothrow(Args&&... args) {
    return std::unique_ptr<T>(new(std::nothrow) T(std::forward<Args>(args)...));
}
```